.github/ISSUE_TEMPLATE/new-release.md

```diff
@@ -18,29 +18,30 @@ Please do not remove items from the checklist
         `git branch release-$MAJ.$MIN main`
   - [ ] An OWNER pushes the new release branch with
         `git push release-$MAJ.$MIN`
-- [ ] Update things like README, deployment templates, docs, configuration, test/e2e flags.
-      Submit a PR against the release branch: <!-- example #211 #214 under Kueue repo -->
-- [ ] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/jobset/releases)
-  - [ ] Write the change log into the draft release.
-  - [ ] Run
-      `make artifacts IMAGE_REGISTRY=registry.k8s.io/jobset GIT_TAG=$VERSION`
-      to generate the artifacts and upload the files in the `artifacts` folder
-      to the draft release.
-- [ ] An OWNER creates a signed tag running
+- [ ] Update `BRANCH_NAME` in the `Makefile` and set `export VERSION=vX.Y.Z`.
+   - [ ] Run `make prepare-release-branch` to update the assets.
+   - [ ] Update the `CHANGELOG-X.Y`
+   - [ ] Submit a PR with the changes against the release branch <!-- TODO (andreyvelich): Add example here>
+- [ ] After the above PR is merged rebase your local branch and create a signed tag running:
      `git tag -s $VERSION`
       and inserts the changelog into the tag description.
       To perform this step, you need [a PGP key registered on github](https://docs.github.com/en/authentication/managing-commit-signature-verification/checking-for-existing-gpg-keys).
 - [ ] An OWNER pushes the tag with
       `git push $VERSION`
   - Triggers prow to build and publish a staging container image
       `gcr.io/k8s-staging-jobset/jobset:$VERSION`
+- [ ] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/jobset/releases)
+  - [ ] Write the changelog into the draft release.
+  - [ ] Run `make artifacts IMAGE_REGISTRY=registry.k8s.io/jobset`
```

Comment:
Previously `make artifacts IMAGE_REGISTRY=registry.k8s.io/jobset GIT_TAG=$VERSION` was required.

Have you verified that we don't actually need GIT_TAG anymore?